OneCommunityGlobal/HGNRest
# Sav - Refactored minHrsMultiple function to check for user hour validity an…

# Description
This PR fixes up the minHrsMultiple badge. Sometimes if users committed hours or weekly tangible hrs are set to 0 it can cause unexpected results and badges to be awarded incorrectly. 
Fixes #11 (Badge Component priority medium

## Mainly changes explained:
Refactor a couple of variables to be more readable
Added a conditional to check if users can get the badge before running the rest of the function
Refactor the assignment of badges or increase of badge count for readability

## How to test:
Check into the current branch
Do `npm install` and `...` to run this PR locally
Log as admin user
Add test user firstName variable to awardNewBadges function to isolate test user
Go to the profile and adjust committed hours in the volunteer times tab
Go to userHelpers.js on the backend
Add fakeHrs variable to replace last week tangible hrs variable
Change fakeHrs and weekly committed hours to test badges
Test badges for x2, x3, x4 and x5 fakeHrs / committed hours
Make sure to run npm run build and restart the backend every time changes are made to backend files

## Screenshots or videos of changes:
![Screenshot from 2023-03-27 10-24-21](https://user-images.githubusercontent.com/98275089/227969986-766a5eab-2918-4b7e-a483-3e45ebfbde63.png)

Here is a video I made talking through my process of testing this bug.

https://user-images.githubusercontent.com/98275089/229658966-b2c63015-810e-461b-9cf2-c8ca2141562a.mp4
Let me know if you have any questions! 

## Note:
If you want to speed the process up or slow it down go to userProfileJobs.js and adjust the cronjob frequency to your liking.
